queries.sql

```diff
@@ -72,3 +72,102 @@ SELECT neutered, MAX(escape_attempts) FROM animals GROUP BY neutered;
 SELECT species, AVG(escape_attempts) FROM animals 
 WHERE date_of_birth BETWEEN '1990-01-01' AND '2000-12-31'
 GROUP BY species;
+
+-- Write queries using JOIN
+
+SELECT * FROM animals
+LEFT JOIN owners
+ON animals.owners_id = owners.owners_id
+WHERE owners.full_name = 'Melody Pond';
+
+SELECT * FROM animals
+LEFT JOIN species
+ON animals.species_id = species.species_id
+WHERE species.name = 'Pokemon';
+
+SELECT 
+owners.full_name AS owners_full_name,
+animals.name AS animals_name
+FROM owners
+LEFT JOIN animals
+ON owners.owners_id = animals.owners_id;
+
+SELECT species.name, COUNT(*)
+FROM animals
+LEFT JOIN  species
+ON animals.species_id = species.species_id
+GROUP BY species.name;
+
+SELECT * FROM animals
+LEFT JOIN owners
+ON animals.owners_id = owners.owners_id
+LEFT JOIN species
+ON animals.species_id = species.species_id
+WHERE owners.full_name = 'Jennifer Orwell'
+AND species.name = 'Digimon';
+
+SELECT animals.* FROM animals
+LEFT JOIN  owners
+ON animals.owners_id = owners.owners_id
+WHERE owners.full_name = 'Dean Winchester'
+AND escape_attempts = 0;
+
+SELECT owners.full_name, COUNT(owners.full_name)
+FROM animals
+JOIN owners
+ON animals.species_id = owners.owners_id
+GROUP BY owners.full_name;
+
+SELECT animals.name, visits.visit_date
+FROM visits
+LEFT JOIN animals ON animals.animals_id = visits.animal_id
+LEFT JOIN vets ON vets.vets_id = visits.vets_id
+WHERE vets.name = 'William Tatcher';
+
+SELECT animals.name
+FROM visits
+JOIN animals ON animals.animals_id = visits.animal_id
+LEFT JOIN vets ON vets.vets_id = visits.vets_id
+WHERE vets.name = 'Stephanie Mendez';
+
+SELECT vets.*,species.name
+FROM vets
+JOIN specializations ON vets.vets_id = specializations.vets_id
+JOIN species ON specializations.species_id = species.species_id;
+
+SELECT visits.visit_date
+FROM visits
+JOIN animals ON animals.animals_id = visits.animal_id
+JOIN vets ON vets.vets_id = visits.vets_id
+WHERE vets.name = 'Stephanie Mendez'
+AND visit_date BETWEEN '2020-04-01' AND '2020-08-30';
+
+SELECT animals.name, COUNT(animals.name)
+FROM visits
+LEFT JOIN animals ON animals.animals_id = visits.animal_id
+GROUP BY animals.name;
+
+SELECT animals.name, visits.visit_date
+FROM visits
+LEFT JOIN animals ON animals.animals_id = visits.animal_id
+LEFT JOIN vets ON vets.vets_id = visits.vets_id
+WHERE vets.name = 'Maisy Smith';
+
+SELECT animals.*, vets.*, visits.visit_date
+FROM visits
+LEFT JOIN animals ON animals.animals_id = visits.animal_id
+LEFT JOIN vets ON vets.vets_id = visits.vets_id;
```

Comment:
- [ ] To get the `Details for the most recent visit` after the LEFT JOIN statement with vets table the results should be ordered and only one entry should be returned. e.g.
```sql
SELECT animals.*, vets.*, visits.visit_date
FROM visits
LEFT JOIN animals ON animals.id = visits.animal_id
LEFT JOIN vets ON vets.id = visits.vet_id
ORDER BY visits.visit_date DESC
LIMIT 1;
```